packages/runtime/src/helpers/files.ts

```diff
@@ -366,7 +366,7 @@ const baseServerReplacements: Array<[string, string]> = [
     `checkIsManualRevalidate(process.env._REVALIDATE_SSG ? { headers: { 'x-prerender-revalidate': this.renderOpts.previewProps.previewModeId } } : req, this.renderOpts.previewProps)`,
   ],
   // ensure ISR 404 pages send the correct SWR cache headers
-  [`private: isPreviewMode || is404Page && cachedData`, `private: isPreviewMode && cachedData`],
+  [`private: isPreviewMode || is404Page`, `private: isPreviewMode || false`],
```

Comment:
Are the `|| false`s necessary here and below? Would `isPreviewMode` be enough?

Reply:
I know it's weird, but they are necessary because of the way we patch/unpatch the files here: https://github.com/netlify/next-runtime/blob/main/packages/runtime/src/helpers/files.ts#L308

Without the `|| false` it will think the file has already been patched because `private: isPreviewMode` already exists in the original file.